src/main/java/com/cleanroommc/groovyscript/compat/mods/naturesaura/Altar.java

```diff
@@ -0,0 +1,167 @@
+package com.cleanroommc.groovyscript.compat.mods.naturesaura;
+
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.compat.mods.ModSupport;
+import com.cleanroommc.groovyscript.helper.Alias;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import de.ellpeck.naturesaura.api.NaturesAuraAPI;
+import de.ellpeck.naturesaura.api.recipes.AltarRecipe;
+import net.minecraft.util.ResourceLocation;
+import org.apache.commons.lang3.tuple.Pair;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+
+@RegistryDescription
+public class Altar extends VirtualizedRegistry<Pair<ResourceLocation, AltarRecipe>> {
+
+    public Altar() {
+        super(Alias.generateOfClass(Altar.class).andGenerate("Infusion"));
+    }
+
+    @RecipeBuilderDescription(example = {
+            @Example(".name('demo').input(item('minecraft:clay')).catalyst(item('minecraft:clay')).output(item('minecraft:diamond')).aura(100).time(100)"),
+            @Example(".name(resource('example:demo')).input(item('minecraft:clay')).output(item('minecraft:gold_ingot') * 8).aura(30).time(5)"),
+            @Example(".input(item('minecraft:gold_ingot')).output(item('minecraft:diamond')).catalyst(item('minecraft:clay')).aura(50).time(100)")
+    })
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    @Override
+    public void onReload() {
+        removeScripted().forEach(x -> NaturesAuraAPI.ALTAR_RECIPES.remove(x.getKey()));
+        restoreFromBackup().forEach(x -> NaturesAuraAPI.ALTAR_RECIPES.put(x.getKey(), x.getValue()));
+    }
+
+    public void add(AltarRecipe recipe) {
+        if (recipe == null) return;
+        addScripted(Pair.of(recipe.name, recipe));
```

Comment:
Does it really need to be a pait if the recipe already stores the name?

Reply:
good point. ive made the change here, but Roots compat uses a similar design for almost all of its compat - will need to circle back through and adjust all of those. out of scope for this PR though.